onnxruntime/core/platform/posix/env.cc

```diff
@@ -49,12 +49,36 @@ class UnmapFileParam {
   size_t len;
 };
 
+/**
+ * @brief Get System Error
+ *
+ * @return a pair of {errno, error message}
+ */
+static std::pair<int, std::string> GetSystemError() {
+  auto e = errno;
+  char buf[1024];
+  const char* msg = "";
+  if (e > 0) {
+#if defined(__GLIBC__) && defined(_GNU_SOURCE) && !defined(__ANDROID__)
```

Comment:
The code was copied from somewhere else. 
According to the man page, the right macro should be:
#ifdef  (_POSIX_C_SOURCE >= 200112L) && !  _GNU_SOURCE
    //strerror_r returns char*
#else
    //strerror_r returns int
#endif